lib/mgmt_msg.c

```diff
@@ -697,11 +696,12 @@ static void msg_client_connect(struct msg_client *client)
 	struct msg_conn *conn = &client->conn;
 	const char *dbgtag = conn->debug ? conn->mstate.idtag : NULL;
 
-	if (!client->short_circuit_ok ||
-	    !msg_client_connect_short_circuit(client))
+	if (!client->short_circuit_ok)
 		conn->fd =
 			mgmt_msg_connect(client->sopath, MSG_CONN_SEND_BUF_SIZE,
 					 MSG_CONN_RECV_BUF_SIZE, dbgtag);
+	else if (msg_client_connect_short_circuit(client))
+		conn->fd = -1;
```

Comment:
Does it really make sense to continue here and reschedule the connection? As you mentioned in Slack it fails only if fd limit is reaches, so rescheduling won't really help as the system is already in an unusable state.

Reply:
Yes, in this case I think it makes sense b/c maybe the user hasn't initialized the server side yet?

Reply:
unless we also want tomake that an assertable condition

Reply:
I mean, maybe we should just `assert(msg_client_connect_short_circuit(client) == 0)` here to give a quick feedback and make the problem visible to an operator.